### PR TITLE
Added support for Probo.ci environments

### PR DESCRIPTION
**Pull request by Auspicus**
Refer to: https://docs.probo.ci/build/environment-variables/

### In this PR
- Added a map from env vars to `Environment` properties
  - `environment.ci = "probo"`
  - `environment.branch = $BRANCH_NAME`
  - `environment.parallelNonce = $BUILD_ID`
  - `environment.commitSha = $COMMIT_REF`
  - `environment.pullRequestNumber` is taken from `$PULL_REQUEST_LINK`
- Added a test to confirm properties are set correctly

### In the mean time
- Anyone running a Probo environment can use the following to get Percy environment variables set without this patch:
  - `export PERCY_TOKEN=[YOUR_PERCY_TOKEN]`
  - `export PERCY_BRANCH=$BRANCH_NAME`
  - `export PERCY_COMMIT=$COMMIT_REF`
  - `export PERCY_PULL_REQUEST=${PULL_REQUEST_LINK##*\/}`
- **Note:** These need to be set in the same Probo "Step" as the Percy command because Probo creates a new shell environment for each step.

### CODE REVIEW

File: test/environment-test.js [hunk 1]
```diff
@@ -641,4 +641,27 @@ COMMIT_MESSAGE:A shiny new feature`);
...
+        COMMIT_REF: 'probo-commit-sha',
+        BRANCH_NAME: 'probo-branch',
+        PULL_REQUEST_LINK: 'https://github.com/owner/repo-name/pull/123',
+      });
+    });
+
+    it('has the correct properties', function() {
+      assert.strictEqual(environment.ci, 'probo');
+      assert.strictEqual(environment.commitSha, 'probo-commit-sha');
+      assert.strictEqual(environment.targetCommitSha, null);
```
Comment (Auspicus) [inline test/environment-test.js]: Not sure if we need to test for null here. I was just following apparent conventions.

**Comment by timhaines** [comment]
@Auspicus Thanks for the PR!  This is looking good.  

CI systems tend to have up to 3 different ways you can run builds:
* builds for a branch with no pull request associated.
* builds for a pull request, without any additional merging
* builds for a pull request, with latest master merged into it.

Sometimes the way the environment variables work change depending on the build type (especially with those merge builds), and often this isn't documented.  i.e sometimes the COMMIT_REF will point to the merge commit SHA instead of the actual commit that we want to associate the build with.

Do you know if Probo supports these different build types?  If so, could you run the `env` command under each one just to sanity check the environment variables remain good for each type?

When I've heard a thumbs up for you, I'll be happy to merge this and ship a new release.

**Comment by Auspicus** [comment]
As far as I can tell Probo only supports spinning up an environment when you create / update a PR and the `COMMIT_REF` for that environment will always point to the `HEAD` of that PR's associated branch.

I'll talk to Probo's support to confirm and get back to you when I hear from them.

**Comment by AlexU-A** [comment]
> As far as I can tell Probo only supports spinning up an environment when you create / update a PR and the `COMMIT_REF` for that environment will always point to the `HEAD` of that PR's associated branch.
> 
> I'll talk to Probo's support to confirm and get back to you when I hear from them.

You are correct that currently you can only create a probo environment based upon pull requests. However, I talked to my cofounder/CTO last week and I think we are close to being able to enable Probo builds for a variety of reasons. I'm hopeful she'll be able to get to it in the next few weeks, but for now Probo will only work via PRs.

**Comment by timhaines** [comment]
Thanks @Auspicus @AlexU-A - I'll publish a new release in a few minutes, once tests have passed again on master.

**Comment by timhaines** [comment]
[Published](https://github.com/percy/percy-js/releases/tag/v3.0.10).  Thanks again!
